File: vibe-lang/benchmark/output/lua-love/openai/orbital_mechanics.lua
local orb_angles = {0.0, 2.094, 4.189}
local orb_radii = {100.0, 160.0, 220.0}
local orb_speeds = {1.5, 1.0, 0.6}

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Orbiting Bodies Simulation")
end

function love.update(dt)
    for i = 1, 3 do
        orb_angles[i] = orb_angles[i] + orb_speeds[i] * dt
    end
end

function love.draw()
    -- Draw the sun
    love.graphics.setColor(1, 1, 0)
    love.graphics.circle("fill", 400, 300, 20)

    -- Draw orbiting bodies and their paths
    for i = 1, 3 do
        -- Draw orbit path
        love.graphics.setColor(0.5, 0.5, 0.5)
        for j = 0, 35 do
            local angle = j * 0.1745
            local px = 400.0 + math.cos(angle) * orb_radii[i]
            local py = 300.0 + math.sin(angle) * orb_radii[i]
            love.graphics.circle("fill", px, py, 1.0)
        end

        -- Calculate and draw the orbiting body
        local x = 400.0 + math.cos(orb_angles[i]) * orb_radii[i]
        local y = 300.0 + math.sin(orb_angles[i]) * orb_radii[i]
        love.graphics.setColor(1, 1, 1)
        love.graphics.circle("fill", x, y, 10)
    end

    -- Display the current angle of body 0
    love.graphics.setColor(1, 1, 1)
    love.graphics.print("Angle: " .. string.format("%.2f", orb_angles[1]), 10, 10)
end

File: benchmark/output/python-pygame/openai/orbital_mechanics.py
import pygame
import sys
import math

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Orbiting Bodies")
clock = pygame.time.Clock()

orb_angles = [0.0, 2.094, 4.189]
orb_radii = [100.0, 160.0, 220.0]
orb_speeds = [1.5, 1.0, 0.6]

font = pygame.font.Font(None, 36)

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    for i in range(3):
        orb_angles[i] += orb_speeds[i] * dt

    screen.fill((0, 0, 0))

    # Draw the sun
    pygame.draw.circle(screen, (255, 255, 0), (400, 300), 20)

    for i in range(3):
        # Calculate position
        x = 400.0 + math.cos(orb_angles[i]) * orb_radii[i]
        y = 300.0 + math.sin(orb_angles[i]) * orb_radii[i]

        # Draw orbiting body
        pygame.draw.circle(screen, (255, 255, 255), (int(x), int(y)), 10)

        # Draw orbit path
        for j in range(36):
            angle = float(j) * 0.1745
            px = 400.0 + math.cos(angle) * orb_radii[i]
            py = 300.0 + math.sin(angle) * orb_radii[i]
            pygame.draw.circle(screen, (100, 100, 100), (int(px), int(py)), 1)

    # Display angle of body 0
    angle_text = font.render("Angle: " + str(orb_angles[0]), True, (255, 255, 255))
    screen.blit(angle_text, (10, 10))

    pygame.display.flip()

pygame.quit()
sys.exit()